widget/edit.html

```diff
@@ -0,0 +1,171 @@
+                    <labe for="customModule" class="control-label">Data Source<span class="text-danger">*</span>
+                        </label>
+                        <select name="customModule" id="customModule" class="form-control"
+                            data-ng-options="module.type as module.name for module in modules"
+                            data-ng-model="config.customModule"
+                            data-ng-change="loadAttributesForCustomModule(config.customModule)" required>
+                            <option value="">Select an Option</option>
+                        </select>
+                </div>
+
+                <div class="form-group" data-ng-if="config.customModule !== undefined">
+
+                    <div class="margin-top-md margin-bottom-md">
+                        <label class="control-label">Filter Record Which Contains The Data<span
+                                class="text-danger">*</span>
+                        </label>
+                        <div data-cs-conditional data-ng-if="fieldsArray.length > 0" data-fields="$parent.fields"
+                            data-reset-field="$parent.fields" data-mode="'queryFilters'"
+                            data-ng-model="$parent.config.query" data-parent-form="editFunnelWidgetForm"
+                            data-enable-expression="(page==='dashboard' || page==='reporting')" data-show-uuid="true"
+                            data-form-name="'editFunnelWidgetForm'"></div>
+                    </div>
+
+                    <div class="form-group" data-ng-if="fieldsArray.length > 0">
+                        <label class="control-label">Select field for data source<span class="text-danger">*</span>
+                        </label>
+                        <select class="form-control"
+                            data-ng-options="field.name as field.title for field in fieldsArray"
+                            data-ng-model="config.customModuleField" required>
+                            <option value="">Select an Option</option>
+                        </select>
+                    </div>
+
+                    <div class="mertics-widget-border padding-top-md padding-bottom-md"></div>
+
+                    <div data-ng-if="config.customModuleField !==undefined">
+
+                        <div class="custom-alert custom-alert-danger margin-bottom-xlg margin-top-md"
+                            data-ng-if="maxlayers === true">
+                            <span
+                                class="glyphicon glyphicon-exclamation-sign error-icon font-size-20 info-icon margin-top-negative-2 padding-left-sm margin-right-lg"></span>
+                            <span>Maximum 4 layers can be added.</span>
+                        </div>
+
+                        <div class="form-group dashed-border-section" data-ng-repeat="item in config.layers"
```

Comment:
layers code is repetitive and can be avoided by using same title field for both module types, add ng-if to second field which needs to be changed in both module types